### PR TITLE
Comments: Update graphics

### PR DESCRIPTION
**Pull request by Copons**
Fixes and updates the graphics of the Comment Management section.

## Changes

- Comment previews are now 2 liners.
Notice that the current "trick" employed here to truncate the text is a simple fade-out on the last line of the content excerpt. This means that its last bit is faded even if there's no need of truncation - and even on 1-line excerpts.
- The search bar is now hidden, as it's slated to m2 milestone.
- On small screens the section doesn't look weird anymore.
- Devdocs show (again) a list of comments after the n-th regression caused by changing the CommentList structure.

## Screenshots

**2-line comments**
screen shot 2017-06-08 at 18 55 23

**`<480px` normal view**
screen shot 2017-06-08 at 18 55 35

**`<480px` bulk edit view**
screen shot 2017-06-08 at 18 55 56

_(please notice that the 100%-wide count "bubble" is apparently (maybe accidentally?) the intended behaviour)_

cc @Automattic/lannister

### CODE REVIEW

**Comment by drw158** [comment]
30% width for the white fade seemed too big, so I changed it to 50px to see what it'd look like. But then, I saw the problem. The text is wrapping, so the second line of text does not reach the fade:

screen shot 2017-06-08 at 2 34 46 pm

Any way we can do white-space: nowrap on the second line only, so that the text goes all the way to right?

**Comment by drw158** [comment]
We can address the mobile view styles once we get the desktop view in a testable state.

Looking good so far!

File: client/blocks/comment-detail/comment-detail-header.jsx [hunk 1]
```diff
@@ -66,16 +70,27 @@ export const CommentDetailHeader = ( {
...
+							{ authorDisplayName }
+						</strong>
+						<span>
+							{ authorUrl }
+						</span>
+					</div>
+					<div className="comment-detail__author-info-element">
+						{ translate( 'on {{postLink/}}', {
+							components: {
+								postLink: <a href={ postUrl }>{ postTitle }</a>,
```
Comment (gwwar) [inline client/blocks/comment-detail/comment-detail-header.jsx]: Maybe an external link? http://calypso.localhost:3000/devdocs/design/external-link

Comment (gwwar) [inline client/blocks/comment-detail/comment-detail-header.jsx]: Minor, but I noticed that clicking on this expands the comment, before redirecting to the site.

**Comment by roundhill** [comment]
Loving the new two line preview, looks good. Where are we at with (dangerously) setting the html to get rid of those ugly tags in the preview?

**Comment by gwwar** [comment]
We need someone to 🔍 https://github.com/Automattic/wp-calypso/issues/14536. I would recommend that we do this as early as possible, just in case this uncovers more work for us.

**Comment by Copons** [comment]
> I noticed one visual issue that only happens in some cases, that we might want to tackle in a follow up PR

@gwwar I know about that issue and I've been struggling for several days to get rid of it.
I actually solved it for the empty content component (that was displaced as well) in https://github.com/Automattic/wp-calypso/pull/14933/commits/9ff138f91a50302fb31c5a7b1cc78e73f6a9ad47, but I couldn't get it to work with the comment list. --But, today I woke up with an idea that might work, gonna try it later on. :)

**Comment by Copons** [comment]
> 30% width for the white fade seemed too big, so I changed it to 50px to see what it'd look like. But then, I saw the problem. The text is wrapping, so the second line of text does not reach the fade:

@drw158  Yup, that's why I went with a fairly large fade.

> Any way we can do white-space: nowrap on the second line only, so that the text goes all the way to right?
> Is it any easier without the implementing the fade?

Unfortunately there's no way (that I know?) to `nowrap` only one line in CSS.
I'm actually thinking of using JS to calculate it, but it would be quite hacky and likely have a big performance hit with many comments (re-calculate and re-render on resize 😨 ).

It might be worth adding `-webkit-line-clamp` on supported browsers and fallback to the fade for anybody else.

**Comment by Copons** [comment]
> We can address the mobile view styles once we get the desktop view in a testable state.

@drw158 True, but try resizing to <480px on the current master! The nav bar looks _really_ bad.
I've just moved stuff around in order to give it a semblance of... normality! 🙂

**Comment by Copons** [comment]
To roundup my latest changes:

- Use `ExternalLink` for the header link.

- An external function will control the link:
  - If we are in bulk edit mode: the link has `preventDefault()`. It doesn't open any link when clicked, it doesn't change color on hover nor has the dotted outline when focused / clicked.
@drw158 is this ok or should I even remove the link color (like, `$gray-text` instead of `$blue-wordpress`) to make it more obvious that it's not clickable?
  - Otherwise: it has `stopPropagation()`. When clicked it opens the link without expanding the comment details.

- In bulk edit mode, the entire comment is clickable.

**Comment by drw158** [comment]
I think using line-clamp is fine, although I'm curious how we do it in notifications.

I've actually been mulling on changing the post link in the comment list item from blue to gray anyway, since that'll make it easier to scan the list and it's not a very important link. Let's change it to gray by default.

**Pull request merged**